### PR TITLE
No match try again

### PR DESCRIPTION
**Pull request by jeffreyquiambao**
- added functionality for host to try swiping again if no match occurred
- same try again functionality should carry over to related stories

### CODE REVIEW

File: wewatch-client/src/views/swiping/swiping-container.js [hunk 1]
```diff
@@ -102,11 +113,16 @@ const SwipingContainer = (props) => {
...
-                                      state: {matchedMovie: matchedMovie}
-                                  }}/>: null }
-          { noMatch ? <Redirect to='/noMatch'/> : null }
+            { match && matchedMovie != null ? <Redirect to={{ 
+                                pathname: '/match',
+                                state: {matchedMovie: matchedMovie}
+                            }}/>: null }
+    { noMatch ? <Redirect to={{
+                                pathname: '/noMatch',
+                                state: {isHost: isHost,
```
Comment (rosakng) [inline wewatch-client/src/views/swiping/swiping-container.js:122]: niceeee 👍

File: wewatch-client/src/components/noMatch/noMatch.js [hunk 2]
```diff
@@ -1,29 +1,70 @@
...
+        setNewRoomId(newRoomId)
+        setTryAgain(true);
+    });
+    return () => { socket.off('tryAgainRedirectUser')};
+    })
     
     return (
         <Layout>
             <div className="noMatchHeroContainer">
+            { tryAgainHost ? <Redirect to={`/lobby?name=${name}&reset=${true}&oldRoomId=${roomId}`}/> : null}
```
Comment (rosakng) [inline wewatch-client/src/components/noMatch/noMatch.js:54]: looks good! I like how you differentiated the try again functionality for both host and non-hosts

File: wewatch-client/src/components/Match/Match.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ import Layout from 'views/layout';
 
 const Match = (props) => {
     const movieData = props.location.state.matchedMovie;
+    console.log(props)
```
Comment (rosakng) [inline wewatch-client/src/components/Match/Match.js:12]: nit: i think we should remove logs that arent showing a changed state or event within the app

File: server/index.js [hunk 4]
```diff
@@ -112,6 +114,23 @@ io.on('connection', (socket) => {
             console.log(`error on dislike event to redis: ${error}`);
         })
     });
+
+    socket.on('try_again_event', ({roomId}) => {
+        console.log("hello")
```
Comment (rosakng) [inline server/index.js:119]: nit: delete console :)

File: wewatch-client/src/components/noMatch/noMatch.js [hunk 5]
```diff
@@ -1,29 +1,70 @@
...
+
 import './noMatch.css';
 import Layout from 'views/layout';
 
-const NoMatch = () => {
+const NoMatch = (props) => {
     const [cancel, setCancel] = useState(false);
+    const [tryAgain, setTryAgain] = useState(false);
+    const [tryAgainHost, setTryAgainHost] = useState(false);
+    const [name, setName] = useState(props.location.state.name);
```
Comment (rosakng) [inline wewatch-client/src/components/noMatch/noMatch.js:13]: is setName being used anywhere in this noMatch page? If not should we just have name as a variable in this component? Let me know your thoughts!

File: server/index.js (hunk 4 above)
Comment (jeffreyquiambao) [reply server/index.js:119]: oops hehe

File: wewatch-client/src/components/noMatch/noMatch.js (hunk 5 above)
Comment (jeffreyquiambao) [reply wewatch-client/src/components/noMatch/noMatch.js:13]: that makes sense, ill just make it a variable!

**Review by billxsheng — APPROVED** [review]
LGTM!

**Review by varoongupta — APPROVED** [review]
Looks good, thanks!

**Pull request merged**